File: app/src/main/java/org/intelehealth/app/shared/FirstLetterUpperCaseInputFilter.java
package org.intelehealth.app.shared;

import android.text.InputFilter;
import android.text.Spanned;
import org.intelehealth.app.utilities.CustomLog;

/**
 * Created by Kaveri Zaware on 28-07-2023
 * email - kaveri@intelehealth.org
 **/
public class FirstLetterUpperCaseInputFilter implements InputFilter {
    private static final String TAG = "FirstLaterFilter";

    @Override
    public CharSequence filter(CharSequence source, int start, int end, Spanned dest, int dstart, int dend) {
        CustomLog.e(TAG, "filter: " + source);
        CustomLog.e(TAG, "Spanned: " + dest);
        CustomLog.e(TAG, "start: " + start);
        CustomLog.e(TAG, "end: " + end);
        CustomLog.e(TAG, "dstart: " + dstart);
        CustomLog.e(TAG, "dend: " + dend);
        CustomLog.e(TAG, "source.length: " + source.length());
        if (dstart == 0 && end == 1) {
            // Convert the first character to uppercase
            char c = source.charAt(0);
            if (Character.isLowerCase(c)) {
                String str = String.valueOf(Character.toUpperCase(c)) + source.subSequence(1, end);
                CustomLog.e(TAG, "toUpperCase: " + str);
                return String.valueOf(Character.toUpperCase(c));
            } else return null;
        }
        return null; // Accept all other input
    }
}

File: app/src/main/java/org/intelehealth/app/ui/filter/FirstLetterUpperCaseInputFilter.kt
package org.intelehealth.app.ui.filter

import android.text.InputFilter
import android.text.Spanned
import org.intelehealth.app.utilities.StringUtils.inputFilter_Others

class FirstLetterUpperCaseInputFilter : InputFilter {
    override fun filter(
        source: CharSequence,
        start: Int,
        end: Int,
        dest: Spanned,
        dstart: Int,
        dend: Int
    ): CharSequence {
        val input = source.subSequence(start, end).toString()
        if (dstart == 0 && source.isNotEmpty()) {
            // Capitalize the first character if it's at the beginning of the text
            val firstChar = source[0].uppercase()
            val restOfInput = if (input.length > 1) input.substring(1) else ""
            return firstChar + restOfInput
        }

        // If the filtered text differs from the original, return it
        return source
    }
}

/*
class FirstLetterUpperCaseInputFilter : InputFilter {
    override fun filter(
        source: CharSequence,
        start: Int,
        end: Int,
        dest: Spanned,
        dstart: Int,
        dend: Int
    ): CharSequence {
        val input = source.subSequence(start, end).toString()
        if (dstart == 0 && source.isNotEmpty()) {
            // Capitalize the first character if it's at the beginning of the text
            val firstChar = source[0].uppercase()
            val restOfInput = if (input.length > 1) input.substring(1) else ""
            val name = firstChar + restOfInput
            return inputFilter_Others.filter(
                name,
                start,
                end,
                dest,
                dstart,
                dend
            )
        }

        // If the filtered text differs from the original, return it
        return inputFilter_Others.filter(
            source,
            start,
            end,
            dest,
            dstart,
            dend
        )
    }

}*/
